tests/integration/test_tls.py

```diff
@@ -154,7 +152,13 @@ async def test_mattermost_db(ops_test: OpsTest) -> None:
 
         # Stop the initial primary.
         logger.info(f"stopping database on {primary}")
-        await run_command_on_unit(ops_test, primary, "/charm/bin/pebble stop postgresql")
+        try:
+            await run_command_on_unit(ops_test, primary, "/charm/bin/pebble stop postgresql")
+        except Exception as e:
+            # pebble stop on juju 2 errors out and leaves dangling PG processes
+            if juju_major_version > 2:
+                raise e
+            await run_command_on_unit(ops_test, primary, "pkill --signal SIGTERM -x postgres")
```

Comment:
On Juju 2 if the `pebble stop` command fails, the Patroni process is terminated, but a dangling Postgresql process remains and prevents Patroni from starting again.

Alternatively, we can do this check and termination in the charm code, but I would prefer not to add Juju 2 specific live code. Thoughts?

Reply:
On 2.9 if patroni stop failed: kill -SIGTERM $postgresql_pid ?

Reply:
> On 2.9 if patroni stop failed: kill -SIGTERM $postgresql_pid ?

Pretty much. We can do this in the ChangeError in the restart check. Question is do we want to add that code for Juju 2's sake?

Reply:
I prefer that we keep this SIGTERM call in the test code and not in the charm code. Even if it is a SIGTERM, I'd prefer either to wait for the database process to gracefully finish by itself or analyze the situation manually before doing any action to avoid any data corruption.